src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
Right, so if `usize` is a `u8`, then `isize` is an `i8` with a maximum value of `127`, and a value of `200` will do Bad Things. Scale up as appropriate for the platform.

Is there a possibility we could add the weak linking to `rlibc` and only depend on it for the appropriate platform(s)? I'm a big fan of laser-focused crates.